src/murfey/server/ispyb.py

```diff
@@ -22,6 +29,30 @@
 )
 
 
+class TransportManager:
+    def __init__(self, transport_type):
+        self.transport = workflows.transport.lookup(transport_type)()
+        self.transport.connect()
+
+    def start_dc(self, message):
+        first_dcgid_list = get_data_collection_group_ids(message["session_id"])
+        self.insert_data_collection_group(message)
+        time.sleep(1)
+        second_dcgid_list = get_data_collection_group_ids(message["session_id"])
+        dcgid = list(set(second_dcgid_list) - set(first_dcgid_list))
+
+        message["ispyb_command"] = "insert_data_collection"
+        message["dcgid"] = dcgid[0]
+        ispyb_message = {"content": "Murfey DC insert", "parameters": message}
+        self.transport.send("ispyb_connector", ispyb_message)
+
+    def insert_data_collection_group(self, message):
+        message["ispyb_command"] = "insert_data_collection_group"
+        ispyb_message = {"content": "Murfey DCG insert", "parameters": message}
+        self.transport.send("ispyb_connector", ispyb_message)
+        return
```

Comment:
Can't do it this way, there are too many assumptions here.
Also there is no need to do it this way: you can send a [multipart message](https://github.com/DiamondLightSource/python-dlstbx/blob/950c89aa9ae0268dc722ee227cbbca36071b27ba/src/dlstbx/services/ispybsvc.py#L996) of "1. Create a DCG, 2. use the output to create a DC"
See the `ispyb-relion` recipe and how it creates, sets, and then uses `ispyb_autoprocprogram_id` for an example